### PR TITLE
feat: Select support maxCount prop

### LINKED ISSUE #6626: Select : why there's no  max / min prop which I can control the length of selected items.
### What problem does this feature solve?
when i use the Select Component, and use mode="Multiple" , I want to limit the length of selected items.  There is no convenient method/props I can use.

### What does the proposed API look like?
min={1} max={5}

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

- ref https://github.com/react-component/select/pull/1012
- ref https://github.com/react-component/select/pull/1015
-  close #6626

### 💡 Background and solution

image

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | feat: Select support maxCount prop |
| 🇨🇳 Chinese | Select 组件支持 maxCount 属性 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -147,7 +147,7 @@
     "rc-rate": "~2.12.0",
     "rc-resize-observer": "^1.4.0",
     "rc-segmented": "~2.2.2",
-    "rc-select": "~14.10.0",
+    "rc-select": "~14.11.0",
```
Comment (li-jia-nan) [inline package.json:150]: 14.11.0 还没发布，先保持 WIP 状态

- wait https://github.com/react-component/select/pull/1012

**Comment by MadCcc** [comment]
避免无关改动。。

File: components/select/demo/maxCount.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+## zh-CN
+
+你可以通过设置 `maxCount` 约束最多可选中的数量，当超出限制时会变成禁止选中状态。
+
+## en-US
+
+You can set the `maxCount` prop to control the max number of items can be selected, When the limit is exceeded, the items will become disabled.
```
Comment (MadCcc) [inline components/select/demo/maxCount.md:7]:
```suggestion
You can set the `maxCount` prop to control the max number of items can be selected。 When the limit is exceeded, the options will become disabled.
```

**Comment by li-jia-nan** [comment]
> 避免无关改动，这些文件重命名可以新开个 PR

done

File: components/select/demo/maxCount.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+## zh-CN
+
+你可以通过设置 `maxCount` 约束最多可选中的数量，当超出限制时会变成禁止选中状态。
+
+## en-US
+
+You can set the `maxCount` prop to control the max number of items can be selected。 When the limit is exceeded, the options will become disabled.
```
Comment (MadCcc) [inline components/select/demo/maxCount.md:7]:
```suggestion
You can set the `maxCount` prop to control the max number of items can be selected. When the limit is exceeded, the options will become disabled.
```

File: components/select/demo/maxCount.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      onChange={setValue}
+      suffixIcon={suffix}
+      placeholder="Please select"
+      options={[
+        { value: 'jack', label: 'Jack' },
+        { value: 'lucy', label: 'Lucy' },
+        { value: 'Yiminghe', label: 'yiminghe' },
+        { value: 'afc163', label: 'afc163' },
+        { value: 'zombie', label: 'zombie' },
+        { value: 'MadCcc', label: 'MadCcc' },
```
Comment (MadCcc) [inline components/select/demo/maxCount.tsx:34]: 个人感觉还是放点虚拟名字好点。。

Comment (li-jia-nan) [reply components/select/demo/maxCount.tsx:34]: 我改一下

File: components/select/index.tsx [hunk 5]
```diff
@@ -263,6 +264,12 @@ const InternalSelect = <
     );
 
     warning.deprecated(!('bordered' in props), 'bordered', 'variant');
+
+    warning(
+      !(typeof maxCount !== 'undefined' && !isMultiple),
+      'usage',
+      'the `maxCount` prop only valid in multiple mode or tags mode',
```
Comment (MadCcc) [inline components/select/index.tsx:271]:
```suggestion
      '`maxCount` only works with mode `multiple` or `tags`',
```

File: components/select/demo/maxCount.tsx (hunk 4 above)
Comment (li-jia-nan) [reply components/select/demo/maxCount.tsx]: done

Comment (li-jia-nan) [reply components/select/demo/maxCount.tsx]: image

**Review by MadCcc — APPROVED** [review]

**Comment by li-jia-nan** [comment]
> 避免无关改动，这些文件重命名可以新开个 PR

- #46759

**Pull request merged**